15_3_Receiving_Bitcoind_Notifications_with_C.md

```diff
@@ -0,0 +1,196 @@
+
+## Install ZMQ
+
+To create a blockchain listener in C you need to install ZMQ following this steps.
+
+  1. Install ZMQ
+  2. Create C program.
+  3. Compile C program.
+  4. Configure bitcoind to allow ZMQ notifications
+  5. Execute listener.
+
+### 1. Install ZMQ
+
+```
+sudo apt-get install libzmq3-dev
+```
+Output
+
+```
+Reading package lists... Done
+Building dependency tree       
+Reading state information... Done
+The following NEW packages will be installed:
+  libzmq3-dev
+0 upgraded, 1 newly installed, 0 to remove and 18 not upgraded.
+Need to get 400 kB of archives.
+After this operation, 2.227 kB of additional disk space will be used.
+Get:1 http://es.archive.ubuntu.com/ubuntu bionic-updates/universe amd64 libzmq3-dev amd64 4.2.5-1ubuntu0.2 [400 kB]
+Fetched 400 kB in 0s (1.114 kB/s)   
+Selecting previously unselected package libzmq3-dev:amd64.
+(Reading database ... 313982 files and directories currently installed.)
+Preparing to unpack .../libzmq3-dev_4.2.5-1ubuntu0.2_amd64.deb ...
+Unpacking libzmq3-dev:amd64 (4.2.5-1ubuntu0.2) ...
+Setting up libzmq3-dev:amd64 (4.2.5-1ubuntu0.2) ...
+Processing triggers for man-db (2.8.3-2ubuntu0.1) ...
+```
+Then 
+
+```
+sudo apt-get install libczmq-dev
+```
+Output:
+
+```
+Reading package lists... Done
+Building dependency tree       
+Reading state information... Done
+The following additional packages will be installed:
+  libczmq4
+The following NEW packages will be installed:
+  libczmq-dev libczmq4
+0 upgraded, 2 newly installed, 0 to remove and 18 not upgraded.
+Need to get 496 kB of archives.
+After this operation, 2.148 kB of additional disk space will be used.
+Do you want to continue? [Y/n] y
+Get:1 http://es.archive.ubuntu.com/ubuntu bionic/universe amd64 libczmq4 amd64 4.1.0-2 [145 kB]
+Get:2 http://es.archive.ubuntu.com/ubuntu bionic/universe amd64 libczmq-dev amd64 4.1.0-2 [351 kB]
+Fetched 496 kB in 0s (1.287 kB/s)    
+Selecting previously unselected package libczmq4:amd64.
+(Reading database ... 314055 files and directories currently installed.)
+Preparing to unpack .../libczmq4_4.1.0-2_amd64.deb ...
+Unpacking libczmq4:amd64 (4.1.0-2) ...
+Selecting previously unselected package libczmq-dev:amd64.
+Preparing to unpack .../libczmq-dev_4.1.0-2_amd64.deb ...
+Unpacking libczmq-dev:amd64 (4.1.0-2) ...
+Setting up libczmq4:amd64 (4.1.0-2) ...
+Setting up libczmq-dev:amd64 (4.1.0-2) ...
+Processing triggers for man-db (2.8.3-2ubuntu0.1) ...
+Processing triggers for libc-bin (2.27-3ubuntu1) ...
+```
+
+### 2. Create C Program
+
```

Comment:
We should describe generally what this program does. I don't think we need to go line by line because it's mostly just a connection program, but saying what it is and what it does would be helpful.

Reply:
I've added a short description what about program does and added examples of topics in parameters.

Reply:
Great.